### ISSUE TITLE
Build fails while checking gradle file

### ISSUE THREAD

**Issue by CSBaum** [nebula-plugins/gradle-lint-plugin#14]
I setup my gradle project to pull the lint plugin in and ran the lintGradle task and got this response:

```
:lintGradle FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':lintGradle'.
> fromIndex(141) > toIndex(140)

```
I setup the plugin to pull in both 0.22 and 0.23 and used the "all-dependency" rule.  If i only use the dependency-* rules it works fine.

I did have to add the jgit & commons-lang packages to the class path

Thanks in advance,
Sean Stallbaum

**Comment by jkschneider** [comment]
@CSBaum Thanks for the report! Could you run `./gradlew lG --stacktrace` and give us the full stacktrace?

**Comment by jkschneider** [comment]
@CSBaum 

> I did have to add the jgit & commons-lang packages to the class path

This has been fixed with 0.24.1

**Comment by CSBaum** [comment]
@jkschneider  Thanks for the quick response, i have attached the output with the stacktrace.  

[lG_stacktrace2.txt](https://github.com/nebula-plugins/gradle-lint-plugin/files/245030/lG_stacktrace2.txt)

**Comment by CSBaum** [comment]
@jkschneider it looks like the jgit dependency has been fixed, but i still need to add the common-lang 2.6 dependency into the build to get it to run.

**Comment by jkschneider** [comment]
@CSBaum Thanks again. Both the commons-lang issue and the original issue about `fromIndex(141) > toIndex(140)` have been fixed with 029fd1c58a3e870068bed6cce7ff77c148010070 and released under 0.25.0.

Much of the turbulence this last few releases stems from how we rewrote the way fixes were tracked and applied. Now the plugin generates a **lint.patch** file in your project's build directory that is just a git-style unified diff patch. This allows us to write rules that affect files other than just build.gradle. For example, we wrote a rule internally that helps us get off a managed gradle wrapper and have teams commit the gradle wrapper while still pushing them along on Gradle versions as new releases come out. To do this, we had to touch .gitignore, the gradle directory (sometimes modifying the file mode to remove committed symlinks), gradle wrapper scripts, etc. The patch-based system allows us to do all of that.

We'll be running the 0.25.0 release against thousands of distinct build at Netflix here shortly, and that should smoke out anything else that might be problematic (if there is anything left).

Once again, I appreciate the early signal of some flaws in this mechanism from you! It's been very helpful.
